ravenframework/Optimizers/GeneticAlgorithm.py

```diff
@@ -776,18 +787,49 @@ def _useRealization(self, info, rlz):
                                               points=self._crossoverPoints)
-
+    elif(self._crossoverProb == "linear"):
+        self._crossoverProb = adaptiveLinearCrossoverProbability(self.getIteration(traj),self.limit)
+        childrenXover = self._crossoverInstance(parents=parents,
+                                              variables=list(self.toBeSampled),
+                                              crossoverProb=self._crossoverProb,
+                                              points=self._crossoverPoints)
+        self._crossoverProb = "linear"
+    elif(self._crossoverProb == "quadratic"):
+        self._crossoverProb = adaptiveQuadraticCrossoverProbability(self.getIteration(traj),self.limit)
+        childrenXover = self._crossoverInstance(parents=parents,
+                                              variables=list(self.toBeSampled),
+                                              crossoverProb=self._crossoverProb,
+                                              points=self._crossoverPoints)
+        self._crossoverProb = "quadratic"
       # 3 @ n: Mutation
       # Perform random directly on childrenCoordinates
+    if(self._mutationProbTpye == "static"):
+      childrenXover = self._crossoverInstance(parents=parents,
+                                            variables=list(self.toBeSampled),
+                                            crossoverProb=self._crossoverProb,
+                                            points=self._crossoverPoints)
+    elif(self._mutationProb == "linear"):
+      self._mutationProb = adaptiveLinearMutationProbability(self.getIteration(traj),self.limit)
+      childrenMutated = self._mutationInstance(offSprings=childrenXover,
+                                              distDict=self.distDict,
+                                              locs=self._mutationLocs,
+                                              mutationProb=self._mutationProb,
+                                              variables=list(self.toBeSampled))
+      self._mutationProb = "linear"
+    elif(self._mutationProb == "quadratic"):
+      self._mutationProb = adaptiveQuadraticmutationProbability(self.getIteration(traj),self.limit)
       childrenMutated = self._mutationInstance(offSprings=childrenXover,
-                                               distDict=self.distDict,
-                                               locs=self._mutationLocs,
-                                               mutationProb=self._mutationProb,
-                                               variables=list(self.toBeSampled))
+                                              distDict=self.distDict,
+                                              locs=self._mutationLocs,
+                                              mutationProb=self._mutationProb,
+                                              variables=list(self.toBeSampled))
+      self._mutationProb = "quadratic"
 
```

Comment:
same here: 1. remove `self._` from lines 818, and 826.
2. remove the lines 824, and 832.
3. Should 813 be a mutation instance instead of a crossover?
4. move 819, 827, and 813 (if I am correct in step 3) and just do it once out of the if statement.
5. After that, there should be another step that should be removed, but I will let you figure it out. :)